src/storage/txn/store.rs

```diff
@@ -127,3 +127,209 @@ impl<'a> StoreScanner<'a> {
+
+    const KEY_PREFIX: &str = "key_prefix";
+    const START_TS: u64 = 10;
+    const COMMIT_TS: u64 = 20;
+    const START_ID: u64 = 1000;
+
+    struct TestStore {
+        keys: Vec<String>,
+        snapshot: Box<Snapshot>,
+        ctx: Context,
+        engine: Box<Engine>,
+    }
+
+    impl TestStore {
+        fn new(key_num: u64) -> TestStore {
+            let engine = engine::new_local_engine(TEMP_DIR, ALL_CFS).unwrap();
+            let keys: Vec<String> =
+                (START_ID..START_ID + key_num).map(|i| format!("{}{}", KEY_PREFIX, i)).collect();
+            let ctx = Context::new();
+            let snapshot = engine.snapshot(&ctx).unwrap();
+            let mut store = TestStore {
+                keys: keys,
+                snapshot: snapshot,
+                ctx: ctx,
+                engine: engine,
+            };
+            store.init_data();
+            store
+        }
+
+        #[inline]
+        fn init_data(&mut self) {
+            let primary_key = format!("{}{}", KEY_PREFIX, START_ID);
+            let pk = primary_key.as_bytes();
+            let mut statistics = self.statistics();
+            // do prewrite.
+            {
+                let mut txn = MvccTxn::new(self.snapshot.as_ref(), &mut statistics, START_TS, None);
+                for key in &self.keys {
+                    let key = key.as_bytes();
+                    txn.prewrite(Mutation::Put((make_key(key), key.to_vec())),
+                                  pk,
+                                  &Options::default())
+                        .unwrap();
+                }
+                self.engine.write(&self.ctx, txn.modifies()).unwrap();
+            }
+            self.refresh_snapshot();
+            // do commit
+            {
+                let mut txn = MvccTxn::new(self.snapshot.as_ref(), &mut statistics, START_TS, None);
+                for key in &self.keys {
+                    let key = key.as_bytes();
+                    txn.commit(&make_key(key), COMMIT_TS).unwrap();
+                }
+                self.engine.write(&self.ctx, txn.modifies()).unwrap();
+            }
+            self.refresh_snapshot();
+        }
+
+        #[inline]
+        fn refresh_snapshot(&mut self) {
+            self.snapshot = self.engine.snapshot(&self.ctx).unwrap()
+        }
+
+        fn store(&self) -> SnapshotStore {
+            SnapshotStore::new(self.snapshot.as_ref(), COMMIT_TS + 1)
+        }
+
+        fn statistics(&self) -> Statistics {
+            Statistics::default()
+        }
+    }
+
+
+    #[test]
+    fn test_snapshot_store_get() {
+        let key_num = 100;
+        let store = TestStore::new(key_num);
+        let snapshot_store = store.store();
+        let mut statistics = store.statistics();
+        for key in &store.keys {
+            let key = key.as_bytes();
+            let data = snapshot_store.get(&make_key(key), &mut statistics).unwrap();
+            assert!(data.is_some());
```

Comment:
Add a readable log message for this assert, so that it's easy to know error happens when getting which key.